2/contrib/s2i/run

```diff
@@ -60,6 +60,14 @@ function create_jenkins_config_from_templates() {
 }
 
 function install_plugins() {
+  mkdir -p ${JENKINS_HOME}/plugins
+  
+  # with the /usr/lib64/jenkins -> /usr/lib/jenkins symlink, don't need to check both directories
+  if [ "$(ls -A /usr/lib/jenkins 2>/dev/null)" ]; then
```

Comment:
nits: the ls -A selects a different set of files to the $(ls in the next line, which is different again to the /usr/lib/jenkins/*hpi below that. Copying *hpi doesn't guarantee that basename ... .hpi will work.  I'd be tempted to:
1) remove the if statement
2) echo "Copying $(ls /usr/lib/jenkins/*.hpi | wc -l) ...
3) for FILENAME in /usr/lib/jenkins/*.hpi ...

Reply:
In your 3), the crux of what you are saying is to do `*.hpi` instead of `*hpi`, right?

Reply:
also, you've got me wondering now if we should ditch the `ls -A` check around `/opt/openshift/plugins` 

I guess the general question is will the for loop blow up if the file expr supplied to the `in` clause does not work.

Will try an experiment locally and report back.

Reply:
sorry - I guess the if check is needed, but it might make more sense if it were if [ "$(ls /usr/lib/jenkins/*.hpi ...

Reply:
`*.hpi` instead of `*hpi`, yes.

Reply:
the answer is that it blows up:

```
gmontero ~ $ ls -la /usr/lib/jenkins
ls: cannot access '/usr/lib/jenkins': No such file or directory
gmontero ~ $ for FILENAME in /usr/lib/jenkins/*.hpi ; do ls -la $FILENAME; done
ls: cannot access '/usr/lib/jenkins/*.hpi': No such file or directory
gmontero ~ $ which stat
/usr/bin/stat
gmontero ~ $ for FILENAME in /usr/lib/jenkins/*.hpi ; do cp --remove-destination $FILENAME ${JENKINS_HOME}/plugins/`basename $FILENAME .hpi`.jpi; done
cp: cannot stat '/usr/lib/jenkins/*.hpi': No such file or directory
gmontero ~ $ 
```

It is not a given that `/usr/lib/jenkins` or `/opt/openshift/plugins` exists.

Rather than using `ls` in the if, how about the use of what is described in https://stackoverflow.com/questions/59838/check-if-a-directory-exists-in-a-shell-script

Reply:
actually based on running jenkins img tests locally, going to switch to https://stackoverflow.com/questions/638975/how-do-i-tell-if-a-regular-file-does-not-exist-in-bash and employ `-f` on `/usr/lib/jenkins/*.hpi`

Reply:
check that  ... `-d` results still ok, though error message which is ignored, `-f` with wildcards not so much

back to exploring ls with * but no -A